pkg/models/plan_payments.go

```diff
@@ -0,0 +1,254 @@
+type NonClaimsBasedPaymentType string
+
+//goland:noinspection ALL
+const (
+	// NonClaimsBasedPaymentTypeAdvancedPayment indicates advanced payment
+	NonClaimsBasedPaymentTypeAdvancedPayment NonClaimsBasedPaymentType = "ADVANCED_PAYMENT"
+	// NonClaimsBasedPaymentTypeBundledEpisodeOfCare indicates bundled episode of care
+	NonClaimsBasedPaymentTypeBundledEpisodeOfCare NonClaimsBasedPaymentType = "BUNDLED_EPISODE_OF_CARE"
+	// NonClaimsBasedPaymentTypeCapitationPopulationBasedFull indicates capitation population based full
+	NonClaimsBasedPaymentTypeCapitationPopulationBasedFull NonClaimsBasedPaymentType = "CAPITATION_POPULATION_BASED_FULL"
+	// NonClaimsBasedPaymentTypeCapitationPopulationBasedPartial indicates capitation population based partial
+	NonClaimsBasedPaymentTypeCapitationPopulationBasedPartial NonClaimsBasedPaymentType = "CAPITATION_POPULATION_BASED_PARTIAL"
+	// NonClaimsBasedPaymentTypeCareCoordinationManagementFee indicates care coordination management fee
+	NonClaimsBasedPaymentTypeCareCoordinationManagementFee NonClaimsBasedPaymentType = "CARE_COORDINATION_MANAGEMENT_FEE"
+	// NonClaimsBasedPaymentTypeGlobalBudget indicates global budget
+	NonClaimsBasedPaymentTypeGlobalBudget NonClaimsBasedPaymentType = "GLOBAL_BUDGET"
+	// NonClaimsBasedPaymentTypeGrants indicates grants
+	NonClaimsBasedPaymentTypeGrants NonClaimsBasedPaymentType = "GRANTS"
+	// NonClaimsBasedPaymentTypeIncentivePayment indicates incentive payment
+	NonClaimsBasedPaymentTypeIncentivePayment NonClaimsBasedPaymentType = "INCENTIVE_PAYMENT"
+	// NonClaimsBasedPaymentTypeMAPDSharedSavings indicates MAPD shared savings
+	NonClaimsBasedPaymentTypeMAPDSharedSavings NonClaimsBasedPaymentType = "MAPD_SHARED_SAVINGS"
+	// NonClaimsBasedPaymentTypeSharedSavings indicates shared savings
+	NonClaimsBasedPaymentTypeSharedSavings NonClaimsBasedPaymentType = "SHARED_SAVINGS"
+	// NonClaimsBasedPaymentTypeOther indicates an option not provided
+	NonClaimsBasedPaymentTypeOther NonClaimsBasedPaymentType = "OTHER"
+)
+
+// PayRecipient is an enumeration of options for this category
+type PayRecipient string
+
+//goland:noinspection ALL
+const (
+	// PayRecipientProviders indicates the pay recipient is a provider
+	PayRecipientProviders PayRecipient = "PROVIDERS"
+	// PayRecipientBeneficiaries indicates the pay recipient is a beneficiary
+	PayRecipientBeneficiaries PayRecipient = "BENEFICIARIES"
+	// PayRecipientParticipants indicates the pay recipient is a participant
+	PayRecipientParticipants PayRecipient = "PARTICIPANTS"
+	// PayRecipientStates indicates the pay recipient is a state
+	PayRecipientStates PayRecipient = "STATES"
+	// PayRecipientOther indicates the pay recipient is not included in the provided options
+	PayRecipientOther PayRecipient = "OTHER"
+)
+
+// PayType is the enumeration of options for this category
+type PayType string
+
+//goland:noinspection ALL
+const (
+	// PayTypeClaimsBasedPayments indicates a claims based payment type
```

Comment:
Side note, you don't have to leave a comment for each enum value, you can do it in the block declaration.